nucypher/blockchain/eth/clients.py

```diff
@@ -254,14 +284,84 @@ def block_number(self) -> BlockNumber:
     def coinbase(self) -> ChecksumAddress:
         return self.w3.eth.coinbase
 
-    def wait_for_receipt(self, transaction_hash: str, timeout: int) -> TxReceipt:
-        receipt = self.w3.eth.waitForTransactionReceipt(transaction_hash=transaction_hash, timeout=timeout)
+    def wait_for_receipt(self,
+                         transaction_hash: str,
+                         timeout: float,
+                         confirmations: int = 0) -> TxReceipt:
```

Comment:
What do you think about changing default value from zero to something greater? Because for real usage zero is extreme. Of course will be configuration at some point, but 0 means not to wait at all and mostly will work with test environments and not with mainnet.
How about put here some value for first attempt and remove default value from other layers?